source/main.cc

```diff
@@ -339,7 +353,20 @@ parse_parameters (const std::string &input_as_string,
   // other processors will be ok as well
   if (dealii::Utilities::MPI::this_mpi_process (MPI_COMM_WORLD) != 0)
     {
+#if DEAL_II_VERSION_GTE(8,5,0)
+      try
+        {
+          prm.parse_input_from_string(input_as_string.c_str());
+        }
+      catch (const dealii::ExceptionBase &e)
+        {
+          success = false;
+          e.print_info(std::cerr);
+          std::cerr << std::endl;
```

Comment:
I am leaving in the variation depending on the Deal.II version with the approach you describe for the new version and testing the return value for the old.